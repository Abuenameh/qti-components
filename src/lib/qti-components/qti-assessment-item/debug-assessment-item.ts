import { html, LitElement } from 'lit';

import { customElement, state } from 'lit/decorators.js';
import { consume, ContextConsumer } from '@lit-labs/context';
import { ItemContext, itemContext } from './qti-assessment-item.context';

@customElement('debug-assessment-item')
export class DebugAssessmentItem extends LitElement {
  // @consume({ context: itemContext, subscribe: true })
  // @state()
  // private _context: Item;
  // connectedCallback(): void {
  //   super.connectedCallback();
  //   // new ContextConsumer(
  //   //   this,
  //   //   itemContext,
  //   //   e => this.dispatchEvent(new CustomEvent('context-changed', { bubbles: true, composed: true, detail: e })),
  //   //   true
  //   // );
  // }
  // render() {
  //   return html` <pre>${JSON.stringify(this._context.variables, null, 2)}</pre> `;
  // }
}
