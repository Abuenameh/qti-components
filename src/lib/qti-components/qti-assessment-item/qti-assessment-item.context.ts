import { createContext } from '@lit-labs/context';
import { VariableDeclaration } from '../qti-utilities/Variables';

export interface ItemContext {
  href?: string;
  identifier: string;
  variables: ReadonlyArray<VariableDeclaration<string | string[]>>;
}

export const itemContext = createContext<ItemContext>('item');
