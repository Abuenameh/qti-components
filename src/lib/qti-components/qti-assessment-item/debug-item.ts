import { html, LitElement } from 'lit';

import { customElement, queryAssignedElements, queryAssignedNodes, state } from 'lit/decorators.js';
import { QtiAssessmentItem } from './qti-assessment-item';
import { register } from 'module';
import { VariableDeclaration } from '../qti-utilities/Variables';

@customElement('debug-item')
export class DebugItem extends LitElement {
  private _assessmentItemRef: QtiAssessmentItem | undefined;

  @state()
  private _items: Array<VariableDeclaration<string | string[]>> = [];

  @state()
  private _savedResponses: ReadonlyArray<VariableDeclaration<string | string[]>> = [];

  private _registerItem = (item: QtiAssessmentItem) => {
    this._assessmentItemRef = item;
    this._assessmentItemRef.context;
  };

  private _contextChanged = () => {
    this._items = [...(this._assessmentItemRef?.context.variables ?? [])];
  };

  private _saveResponse = () => {
    this._savedResponses = this._assessmentItemRef.context.variables;
  };

  private _loadResponse = () => {
    this._assessmentItemRef.context = { ...this._assessmentItemRef.context, variables: this._savedResponses };
  };
  render() {
    return html`
      <pre>${JSON.stringify(this._items, null, 2)}</pre>
      <button @click=${this._saveResponse}>Save</button>
      <button @click=${this._loadResponse}>Load</button>

      <slot
        @item-connected=${({ detail }) => this._registerItem(detail)}
        @context-changed=${() => this._contextChanged()}
      >
      </slot>
      <pre>${JSON.stringify(this._savedResponses, null, 2)}</pre>
    `;
  }
}

// @queryAssignedElements({ slot: 'list', flatten: true, selector: 'qti-assessment-item' })
// listItems!: Array<QtiAssessmentItem>;

// @slotchange=${(e: Event) => console.log('slotchange', e, this.listItems)}
