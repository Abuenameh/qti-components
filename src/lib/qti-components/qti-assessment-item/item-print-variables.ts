import { consume } from '@lit/context';
import { LitElement, html } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { ItemContext, itemContext } from './qti-assessment-item.context';

@customElement('item-print-variables')
export class ItemPrintVariables extends LitElement {
  @consume({ context: itemContext, subscribe: true })
  @state()
  public itemContext?: ItemContext;

  render() {
    return html` <pre>${JSON.stringify(this.itemContext, null, 2)}</pre> `;
  }
}
