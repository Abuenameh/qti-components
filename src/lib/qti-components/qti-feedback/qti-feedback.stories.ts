import { html } from 'lit';
import { createRef, ref } from 'lit/directives/ref.js';

import { action } from '@storybook/addon-actions';
import { QtiAssessmentItem } from '../qti-assessment-item/qti-assessment-item';

import { expect, userEvent, within } from '@storybook/test';
import { StoryObj } from '@storybook/web-components';
import '../index';

type Story = StoryObj; // <Props>;

export default {
  component: 'qti-feedback-inline'
};

const InlineFeedbackRender = {
  render: args => {
    const assessmentItemRef = createRef<QtiAssessmentItem>();

    const processResponse = () => {
      assessmentItemRef.value?.processResponse();
    };
    return html`
      <qti-assessment-item
        @qti-outcome-changed="${e => {
          action(JSON.stringify(e.detail))();
        }}"
        @qti-interaction-changed="${e => {
          processResponse();
          action(JSON.stringify(e.detail))();
        }}"
        identifier="blah"
        ${ref(assessmentItemRef)}
      >
        <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="identifier">
          <qti-correct-response>
            <qti-value>true</qti-value>
          </qti-correct-response>
        </qti-response-declaration>
        <!--￼Define a feedback variable; its baseType is "identifier" so that it can contain the identifier 
        of the feedback message-->
        <qti-outcome-declaration
          identifier="FEEDBACK"
          cardinality="single"
          base-type="identifier"
        ></qti-outcome-declaration>
        <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float" normal-maximum="10.0">
          <qti-default-value>
            <qti-value>0</qti-value>
          </qti-default-value>
        </qti-outcome-declaration>
        <qti-outcome-declaration identifier="MAXSCORE" cardinality="single" base-type="float">
          <qti-default-value>
            <qti-value>10.0</qti-value>
          </qti-default-value>
        </qti-outcome-declaration>
        <qti-item-body>
          <!--￼￼￼￼The response variable RESPONSE will hold the candidate's input-->
          <qti-choice-interaction response-identifier="RESPONSE" shuffle="false" max-choices="1">
            <qti-prompt>Sigmund Freud and Carl Jung both belong to the psychoanalytic school of psychology.</qti-prompt>
            <qti-simple-choice data-testid="choice" identifier="true" fixed="true"
              >True
              <!--￼￼￼The feedbackInline elements are each given the same identifier as the 
                    corresponding option.-->
              <qti-feedback-inline
                data-testid="feedback"
                outcome-identifier="FEEDBACK"
                identifier="true"
                show-hide="show"
                >That's correct</qti-feedback-inline
              ></qti-simple-choice
            >
            <qti-simple-choice identifier="false" fixed="true"
              >False
              <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="false" show-hide="show"
                >That's not correct</qti-feedback-inline
              ></qti-simple-choice
            >
          </qti-choice-interaction>
        </qti-item-body>
        <qti-response-processing>
          <!--￼This time, FEEDBACK is given the value of the identifier of the option which was selected.-->
          <qti-set-outcome-value identifier="FEEDBACK">
            <qti-variable identifier="RESPONSE"></qti-variable>
          </qti-set-outcome-value>
          <qti-response-condition>
            <qti-response-if>
              <qti-match>
                <qti-variable identifier="RESPONSE"></qti-variable>
                <qti-correct identifier="RESPONSE"></qti-correct>
              </qti-match>
              <qti-set-outcome-value identifier="SCORE">
                <qti-variable identifier="MAXSCORE"></qti-variable>
              </qti-set-outcome-value>
            </qti-response-if>
          </qti-response-condition>
        </qti-response-processing>
      </qti-assessment-item>
    `;
  }
};

export const InlineFeedback: Story = {
  render: InlineFeedbackRender.render,
  args: {},
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    const choiceElement = canvas.getByTestId('choice');
    const isFeedbackVisible = canvas.getByTestId('feedback').shadowRoot.querySelector('slot');

    await userEvent.click(choiceElement);
    expect(isFeedbackVisible).toHaveClass('on');
    await userEvent.click(choiceElement);
    expect(isFeedbackVisible).toHaveClass('off');
  }
};

export const ModalFeedback = () => {
  const assessmentItemRef = createRef<QtiAssessmentItem>();

  const processResponse = () => {
    assessmentItemRef.value?.processResponse();
  };

  return html` <qti-assessment-item
    @qti-outcome-changed="${e => {
      action(JSON.stringify(e.detail))();
    }}"
    @qti-interaction-changed="${e => {
      processResponse();
      action(JSON.stringify(e.detail))();
    }}"
    identifier="blah"
    ${ref(assessmentItemRef)}
  >
    <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="identifier">
      <qti-correct-response>
        <qti-value>true</qti-value>
      </qti-correct-response>
    </qti-response-declaration>
    <qti-outcome-declaration
      identifier="FEEDBACK"
      cardinality="single"
      base-type="identifier"
    ></qti-outcome-declaration>
    <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float" normal-maximum="10.0">
      <qti-default-value>
        <qti-value>0</qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="MAXSCORE" cardinality="single" base-type="float">
      <qti-default-value>
        <qti-value>10.0</qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-choice-interaction response-identifier="RESPONSE" shuffle="false" max-choices="1">
        <qti-prompt>Sigmund Freud and Carl Jung both belong to the psychoanalytic school of psychology.</qti-prompt>
        <qti-simple-choice identifier="true" fixed="true">True </qti-simple-choice>
        <qti-simple-choice identifier="false" fixed="true">False </qti-simple-choice>
      </qti-choice-interaction>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="FEEDBACK">
        <qti-variable identifier="RESPONSE"></qti-variable>
      </qti-set-outcome-value>
      <qti-response-condition>
        <qti-response-if>
          <qti-match>
            <qti-variable identifier="RESPONSE"></qti-variable>
            <qti-correct identifier="RESPONSE"></qti-correct>
          </qti-match>
          <qti-set-outcome-value identifier="SCORE">
            <qti-variable identifier="MAXSCORE"></qti-variable>
          </qti-set-outcome-value>
        </qti-response-if>
      </qti-response-condition>
    </qti-response-processing>

    <qti-modal-feedback outcome-identifier="FEEDBACK" show-hide="show" identifier="true">
      <qti-content-body>correct</qti-content-body>
    </qti-modal-feedback>
    <qti-modal-feedback outcome-identifier="FEEDBACK" show-hide="show" identifier="false">
      <qti-content-body>incorrect</qti-content-body>
    </qti-modal-feedback>
  </qti-assessment-item>`;
};

export const DirectFeedback = () => {
  const assessmentItemRef = createRef<QtiAssessmentItem>();

  const processResponse = () => {
    assessmentItemRef.value?.processResponse();
  };

  return html`
    <qti-assessment-item
      @qti-outcome-changed="${e => {
        action(JSON.stringify(e.detail))();
      }}"
      @qti-interaction-changed="${e => {
        console.log('qti-interaction-changed: feedback ');
        processResponse();
        action(JSON.stringify(e.detail))();
      }}"
      identifier="blah"
      ${ref(assessmentItemRef)}
    >
      <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="identifier">
        <qti-correct-response>
          <qti-value>A</qti-value>
        </qti-correct-response>
      </qti-response-declaration>
      <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float"></qti-outcome-declaration>
      <qti-outcome-declaration
        identifier="FEEDBACK"
        cardinality="single"
        base-type="identifier"
      ></qti-outcome-declaration>
      <qti-item-body>
        <qti-choice-interaction class="qti-input-control-hidden" response-identifier="RESPONSE" max-choices="1">
          <qti-prompt> Wat betekent stuk? </qti-prompt>
          <qti-simple-choice identifier="A">
            Een deel
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="A" show-hide="show"
              >goed</qti-feedback-inline
            >
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="B" show-hide="show"
              >goed</qti-feedback-inline
            >
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="C" show-hide="show"
              >goed</qti-feedback-inline
            >
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="D" show-hide="show"
              >goed</qti-feedback-inline
            >
          </qti-simple-choice>
          <qti-simple-choice identifier="B"
            >Een gat
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="B" show-hide="show"
              >fout</qti-feedback-inline
            >
          </qti-simple-choice>
          <qti-simple-choice identifier="C"
            >Een krant
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="C" show-hide="show"
              >fout</qti-feedback-inline
            >
          </qti-simple-choice>
          <qti-simple-choice identifier="D">
            Een steen
            <qti-feedback-inline outcome-identifier="FEEDBACK" identifier="D" show-hide="show"
              >fout</qti-feedback-inline
            >
          </qti-simple-choice>
        </qti-choice-interaction>
      </qti-item-body>

      <qti-response-processing>
        <qti-set-outcome-value identifier="FEEDBACK">
          <qti-variable identifier="RESPONSE"></qti-variable>
        </qti-set-outcome-value>

        <!-- <qti-response-condition>
            <qti-response-if>
              <qti-match>
                <qti-variable identifier="RESPONSE"></qti-variable>
                <qti-correct identifier="RESPONSE"></qti-correct>
              </qti-match>
              <qti-set-outcome-value identifier="SCORE">
                <qti-base-value base-type="float">1</qti-base-value>
              </qti-set-outcome-value>
            </qti-response-if>
            <qti-response-else>
              <qti-set-outcome-value identifier="SCORE">
                <qti-base-value base-type="float">0</qti-base-value>
              </qti-set-outcome-value>
            </qti-response-else>
          </qti-response-condition> -->
      </qti-response-processing>
    </qti-assessment-item>
  `;
};

export const AdaptiveFeedback = () => {
  const assessmentItemRef = createRef<QtiAssessmentItem>();

  const processResponse = () => {
    assessmentItemRef.value?.processResponse();
  };

  return html`
    <qti-assessment-item
      @qti-outcome-changed="${e => {
        action(JSON.stringify(e.detail))();
      }}"
      @qti-interaction-changed="${e => {
        processResponse();
        action(JSON.stringify(e.detail))();
      }}"
      identifier="kringloop1"
      ${ref(assessmentItemRef)}
    >
      <qti-response-declaration identifier="RESPONSE" cardinality="single" base-type="float">
        <qti-correct-response interpretation="562 kilo">
          <qti-value>562</qti-value>
        </qti-correct-response>
      </qti-response-declaration>
      <qti-outcome-declaration identifier="SCORE" cardinality="single" base-type="float">
        <qti-default-value>
          <qti-value>0</qti-value>
        </qti-default-value>
      </qti-outcome-declaration>
      <qti-outcome-declaration
        identifier="FEEDBACK"
        cardinality="single"
        base-type="identifier"
      ></qti-outcome-declaration>
      <qti-item-body xml:lang="nl-NL">
        <div class="qti-layout-row">
          <div class="qti-layout-col12">
            <p>
              <strong>Hoeveel kilo afval werd er 2021 in Nederland per persoon weggegooid?</strong>
            </p>
            <div>
              <qti-slider-interaction
                response-identifier="RESPONSE"
                lower-bound="0"
                step="100"
                upper-bound="1000"
              ></qti-slider-interaction>
              <qti-feedback-block
                id="feedbackblock-correct-exact"
                identifier="correct-exact"
                outcome-identifier="FEEDBACK"
                show-hide="show"
              >
                Helemaal goed! Dat is net zoveel als het gewicht van een groot paard! 🐎
              </qti-feedback-block>
              <qti-feedback-block
                id="feedbackblock-correct"
                identifier="correct"
                outcome-identifier="FEEDBACK"
                show-hide="show"
              >
                Bijna! We rekenen het goed! 562 kilo 😮 Dat is net zoveel als het gewicht van een groot paard!🐎
              </qti-feedback-block>
              <qti-feedback-block
                id="feedbackblock-incorrect-less"
                identifier="incorrect-less"
                outcome-identifier="FEEDBACK"
                show-hide="show"
              >
                Helaas het is nog meer! 562 kilo 😮! Dat is net zoveel als het gewicht van een groot paard 🐎
              </qti-feedback-block>
              <qti-feedback-block
                id="feedbackblock-incorrect-more"
                identifier="incorrect-more"
                outcome-identifier="FEEDBACK"
                show-hide="show"
              >
                Dat is niet goed, gelukkig is het minder, maar toch 562 kilo! Dat is net zoveel als het gewicht van een
                groot paard!🐎
              </qti-feedback-block>
            </div>
          </div>
        </div>
      </qti-item-body>
      <qti-response-processing>
        <qti-response-condition>
          <qti-response-if>
            <qti-and>
              <qti-gte>
                <qti-variable identifier="RESPONSE"></qti-variable>
                <qti-base-value base-type="float">500</qti-base-value>
              </qti-gte>
              <qti-lte>
                <qti-variable identifier="RESPONSE"></qti-variable>
                <qti-base-value base-type="float">600</qti-base-value>
              </qti-lte>
            </qti-and>
            <qti-set-outcome-value identifier="SCORE">
              <qti-base-value base-type="float">1</qti-base-value>
            </qti-set-outcome-value>
            <qti-response-condition>
              <qti-response-if>
                <qti-equal>
                  <qti-variable identifier="RESPONSE"></qti-variable>
                  <qti-correct identifier="RESPONSE"></qti-correct>
                </qti-equal>
                <qti-set-outcome-value identifier="FEEDBACK">
                  <qti-base-value base-type="identifier">correct-exact</qti-base-value>
                </qti-set-outcome-value>
              </qti-response-if>
              <qti-response-else>
                <qti-set-outcome-value identifier="FEEDBACK">
                  <qti-base-value base-type="string">correct</qti-base-value>
                </qti-set-outcome-value>
              </qti-response-else>
            </qti-response-condition>
          </qti-response-if>
          <qti-response-else>
            <qti-set-outcome-value identifier="SCORE">
              <qti-base-value base-type="float">0</qti-base-value>
            </qti-set-outcome-value>
            <qti-response-condition>
              <qti-response-if>
                <qti-lt>
                  <qti-variable identifier="RESPONSE"></qti-variable>
                  <qti-base-value base-type="float">500</qti-base-value>
                </qti-lt>
                <qti-set-outcome-value identifier="FEEDBACK">
                  <qti-base-value base-type="string">incorrect-less</qti-base-value>
                </qti-set-outcome-value>
              </qti-response-if>
              <qti-response-else>
                <qti-set-outcome-value identifier="FEEDBACK">
                  <qti-base-value base-type="string">incorrect-more</qti-base-value>
                </qti-set-outcome-value>
              </qti-response-else>
            </qti-response-condition>
          </qti-response-else>
        </qti-response-condition>
      </qti-response-processing>
    </qti-assessment-item>
  `;
};
