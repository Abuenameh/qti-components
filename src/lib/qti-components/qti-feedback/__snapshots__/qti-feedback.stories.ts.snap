// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/qti-components/qti-feedback inline smoke-test 1`] = `
<div id="root-inner">
  <qti-assessment-item identifier="blah">
    <qti-response-declaration identifier="RESPONSE"
                              cardinality="single"
                              base-type="identifier"
    >
      <qti-correct-response>
      </qti-correct-response>
    </qti-response-declaration>
    <qti-outcome-declaration identifier="FEEDBACK"
                             cardinality="single"
                             base-type="identifier"
    >
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="SCORE"
                             cardinality="single"
                             base-type="float"
                             normal-maximum="10.0"
    >
      <qti-default-value>
        <qti-value>
          0
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="MAXSCORE"
                             cardinality="single"
                             base-type="float"
    >
      <qti-default-value>
        <qti-value>
          10.0
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-choice-interaction response-identifier="RESPONSE"
                              shuffle="false"
                              max-choices="1"
      >
        <qti-prompt slot="prompt">
          Sigmund Freud and Carl Jung both belong to the psychoanalytic school of psychology.
        </qti-prompt>
        <qti-simple-choice identifier="true"
                           fixed="true"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          True
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="true"
                               show-hide="show"
          >
            That's correct
          </qti-feedback-inline>
        </qti-simple-choice>
        <qti-simple-choice identifier="false"
                           fixed="true"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          False
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="false"
                               show-hide="show"
          >
            That's not correct
          </qti-feedback-inline>
        </qti-simple-choice>
      </qti-choice-interaction>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="FEEDBACK">
        <qti-variable identifier="RESPONSE">
        </qti-variable>
      </qti-set-outcome-value>
      <qti-response-condition>
        <qti-response-if>
          <qti-match>
            <qti-variable identifier="RESPONSE">
            </qti-variable>
            <qti-correct identifier="RESPONSE">
            </qti-correct>
          </qti-match>
          <qti-set-outcome-value identifier="SCORE">
            <qti-variable identifier="MAXSCORE">
            </qti-variable>
          </qti-set-outcome-value>
        </qti-response-if>
      </qti-response-condition>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;

exports[`lib/qti-components/qti-feedback kringloop smoke-test 1`] = `
<div id="root-inner">
  <qti-assessment-item identifier="kringloop1">
    <qti-response-declaration identifier="RESPONSE"
                              cardinality="single"
                              base-type="float"
    >
      <qti-correct-response interpretation="562 kilo">
      </qti-correct-response>
    </qti-response-declaration>
    <qti-outcome-declaration identifier="SCORE"
                             cardinality="single"
                             base-type="float"
    >
      <qti-default-value>
        <qti-value>
          0
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="FEEDBACK"
                             cardinality="single"
                             base-type="identifier"
    >
    </qti-outcome-declaration>
    <qti-item-body xml:lang="nl-NL">
      <div class="qti-layout-row">
        <div class="qti-layout-col12">
          <p>
            <strong>
              Hoeveel kilo afval werd er 2021 in Nederland per persoon weggegooid?
            </strong>
          </p>
          <div>
            <qti-slider-interaction response-identifier="RESPONSE"
                                    lower-bound="0"
                                    step="100"
                                    upper-bound="1000"
                                    tabindex="0"
                                    role="slider"
                                    aria-valuenow="0"
                                    style="--min: 0; --step: 1; --max: 1000; --value-percentage: 0%;"
            >
            </qti-slider-interaction>
            <qti-feedback-block id="feedbackblock-correct-exact"
                                identifier="correct-exact"
                                outcome-identifier="FEEDBACK"
                                show-hide="show"
            >
              Helemaal goed! Dat is net zoveel als het gewicht van een groot paard! üêé
            </qti-feedback-block>
            <qti-feedback-block id="feedbackblock-correct"
                                identifier="correct"
                                outcome-identifier="FEEDBACK"
                                show-hide="show"
            >
              Bijna! We rekenen het goed! 562 kilo üòÆ Dat is net zoveel als het gewicht van een groot paard!üêé
            </qti-feedback-block>
            <qti-feedback-block id="feedbackblock-incorrect-less"
                                identifier="incorrect-less"
                                outcome-identifier="FEEDBACK"
                                show-hide="show"
            >
              Helaas het is nog meer! 562 kilo üòÆ! Dat is net zoveel als het gewicht van een groot paard üêé
            </qti-feedback-block>
            <qti-feedback-block id="feedbackblock-incorrect-more"
                                identifier="incorrect-more"
                                outcome-identifier="FEEDBACK"
                                show-hide="show"
            >
              Dat is niet goed, gelukkig is het minder, maar toch 562 kilo! Dat is net zoveel als het gewicht van een
                groot paard!üêé
            </qti-feedback-block>
          </div>
        </div>
      </div>
    </qti-item-body>
    <qti-response-processing>
      <qti-response-condition>
        <qti-response-if>
          <qti-and>
            <qti-gte>
              <qti-variable identifier="RESPONSE">
              </qti-variable>
              <qti-base-value base-type="float">
                500
              </qti-base-value>
            </qti-gte>
            <qti-lte>
              <qti-variable identifier="RESPONSE">
              </qti-variable>
              <qti-base-value base-type="float">
                600
              </qti-base-value>
            </qti-lte>
          </qti-and>
          <qti-set-outcome-value identifier="SCORE">
            <qti-base-value base-type="float">
              1
            </qti-base-value>
          </qti-set-outcome-value>
          <qti-response-condition>
            <qti-response-if>
              <qti-equal>
                <qti-variable identifier="RESPONSE">
                </qti-variable>
                <qti-correct identifier="RESPONSE">
                </qti-correct>
              </qti-equal>
              <qti-set-outcome-value identifier="FEEDBACK">
                <qti-base-value base-type="identifier">
                  correct-exact
                </qti-base-value>
              </qti-set-outcome-value>
            </qti-response-if>
            <qti-response-else>
              <qti-set-outcome-value identifier="FEEDBACK">
                <qti-base-value base-type="string">
                  correct
                </qti-base-value>
              </qti-set-outcome-value>
            </qti-response-else>
          </qti-response-condition>
        </qti-response-if>
        <qti-response-else>
          <qti-set-outcome-value identifier="SCORE">
            <qti-base-value base-type="float">
              0
            </qti-base-value>
          </qti-set-outcome-value>
          <qti-response-condition>
            <qti-response-if>
              <qti-lt>
                <qti-variable identifier="RESPONSE">
                </qti-variable>
                <qti-base-value base-type="float">
                  500
                </qti-base-value>
              </qti-lt>
              <qti-set-outcome-value identifier="FEEDBACK">
                <qti-base-value base-type="string">
                  incorrect-less
                </qti-base-value>
              </qti-set-outcome-value>
            </qti-response-if>
            <qti-response-else>
              <qti-set-outcome-value identifier="FEEDBACK">
                <qti-base-value base-type="string">
                  incorrect-more
                </qti-base-value>
              </qti-set-outcome-value>
            </qti-response-else>
          </qti-response-condition>
        </qti-response-else>
      </qti-response-condition>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;

exports[`lib/qti-components/qti-feedback modal smoke-test 1`] = `
<div id="root-inner">
  <qti-assessment-item identifier="blah">
    <qti-response-declaration identifier="RESPONSE"
                              cardinality="single"
                              base-type="identifier"
    >
      <qti-correct-response>
      </qti-correct-response>
    </qti-response-declaration>
    <qti-outcome-declaration identifier="FEEDBACK"
                             cardinality="single"
                             base-type="identifier"
    >
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="SCORE"
                             cardinality="single"
                             base-type="float"
                             normal-maximum="10.0"
    >
      <qti-default-value>
        <qti-value>
          0
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="MAXSCORE"
                             cardinality="single"
                             base-type="float"
    >
      <qti-default-value>
        <qti-value>
          10.0
        </qti-value>
      </qti-default-value>
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-choice-interaction response-identifier="RESPONSE"
                              shuffle="false"
                              max-choices="1"
      >
        <qti-prompt slot="prompt">
          Sigmund Freud and Carl Jung both belong to the psychoanalytic school of psychology.
        </qti-prompt>
        <qti-simple-choice identifier="true"
                           fixed="true"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          True
        </qti-simple-choice>
        <qti-simple-choice identifier="false"
                           fixed="true"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          False
        </qti-simple-choice>
      </qti-choice-interaction>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="FEEDBACK">
        <qti-variable identifier="RESPONSE">
        </qti-variable>
      </qti-set-outcome-value>
      <qti-response-condition>
        <qti-response-if>
          <qti-match>
            <qti-variable identifier="RESPONSE">
            </qti-variable>
            <qti-correct identifier="RESPONSE">
            </qti-correct>
          </qti-match>
          <qti-set-outcome-value identifier="SCORE">
            <qti-variable identifier="MAXSCORE">
            </qti-variable>
          </qti-set-outcome-value>
        </qti-response-if>
      </qti-response-condition>
    </qti-response-processing>
    <qti-modal-feedback outcome-identifier="FEEDBACK"
                        show-hide="show"
                        identifier="true"
    >
      <qti-content-body>
        correct
      </qti-content-body>
    </qti-modal-feedback>
    <qti-modal-feedback outcome-identifier="FEEDBACK"
                        show-hide="show"
                        identifier="false"
    >
      <qti-content-body>
        incorrect
      </qti-content-body>
    </qti-modal-feedback>
  </qti-assessment-item>
</div>
`;

exports[`lib/qti-components/qti-feedback woordmars smoke-test 1`] = `
<div id="root-inner">
  <qti-assessment-item 2
                       identifier="blah"
  >
    <qti-response-declaration identifier="RESPONSE"
                              cardinality="single"
                              base-type="identifier"
    >
      <qti-correct-response>
      </qti-correct-response>
    </qti-response-declaration>
    <qti-outcome-declaration identifier="SCORE"
                             cardinality="single"
                             base-type="float"
    >
    </qti-outcome-declaration>
    <qti-outcome-declaration identifier="FEEDBACK"
                             cardinality="single"
                             base-type="identifier"
    >
    </qti-outcome-declaration>
    <qti-item-body>
      <qti-choice-interaction class="qti-input-control-hidden"
                              response-identifier="RESPONSE"
                              max-choices="1"
      >
        <qti-prompt slot="prompt">
          Wat betekent stuk?
        </qti-prompt>
        <qti-simple-choice identifier="A"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          Een deel
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="A"
                               show-hide="show"
          >
            goed
          </qti-feedback-inline>
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="B"
                               show-hide="show"
          >
            goed
          </qti-feedback-inline>
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="C"
                               show-hide="show"
          >
            goed
          </qti-feedback-inline>
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="D"
                               show-hide="show"
          >
            goed
          </qti-feedback-inline>
        </qti-simple-choice>
        <qti-simple-choice identifier="B"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          Een gat
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="B"
                               show-hide="show"
          >
            fout
          </qti-feedback-inline>
        </qti-simple-choice>
        <qti-simple-choice identifier="C"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          Een krant
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="C"
                               show-hide="show"
          >
            fout
          </qti-feedback-inline>
        </qti-simple-choice>
        <qti-simple-choice identifier="D"
                           role="radio"
                           tabindex="0"
                           aria-disabled="false"
                           aria-readonly="false"
                           aria-checked="false"
        >
          Een steen
          <qti-feedback-inline outcome-identifier="FEEDBACK"
                               identifier="D"
                               show-hide="show"
          >
            fout
          </qti-feedback-inline>
        </qti-simple-choice>
      </qti-choice-interaction>
    </qti-item-body>
    <qti-response-processing>
      <qti-set-outcome-value identifier="FEEDBACK">
        <qti-variable identifier="RESPONSE">
        </qti-variable>
      </qti-set-outcome-value>
      <qti-response-condition>
        <qti-response-if>
          <qti-match>
            <qti-variable identifier="RESPONSE">
            </qti-variable>
            <qti-correct identifier="RESPONSE">
            </qti-correct>
          </qti-match>
          <qti-set-outcome-value identifier="SCORE">
            <qti-base-value base-type="float">
              1
            </qti-base-value>
          </qti-set-outcome-value>
        </qti-response-if>
        <qti-response-else>
          <qti-set-outcome-value identifier="SCORE">
            <qti-base-value base-type="float">
              0
            </qti-base-value>
          </qti-set-outcome-value>
        </qti-response-else>
      </qti-response-condition>
    </qti-response-processing>
  </qti-assessment-item>
</div>
`;
